plugins/gitstatus.lua

```diff
@@ -39,6 +47,17 @@ function TreeView:get_item_text(item, active, hovered)
   return text, font, color
 end
 
+if config.plugins.gitstatus.color_icons then
+  -- Override TreeView's get_item_icon to add modification color
+  local treeview_get_item_icon = TreeView.get_item_icon
+  function TreeView:get_item_icon(item, active, hovered)
+    local character, font, color = treeview_get_item_icon(self, item, active, hovered)
+    if cached_color_for_item[item.abs_filename] then
+      color = cached_color_for_item[item.abs_filename]
+    end
+    return character, font, color
+  end
+end
```

Comment:
What do you think about something like:
```suggestion
-- Override TreeView's get_item_icon to add modification color
local treeview_get_item_icon = TreeView.get_item_icon
function TreeView:get_item_icon(item, active, hovered)
  local character, font, color = treeview_get_item_icon(self, item, active, hovered)
  if config.plugins.gitstatus and config.plugins.gitstatus.color_icons and cached_color_for_item[item.abs_filename] then
    color = cached_color_for_item[item.abs_filename]
  end
  return character, font, color
end
```

This way changing the option at runtime will apply immediately without a restart.

Reply:
Although I like this approach, the first solution would override `TreeView:get_item_icon` once and would not require checking also `config.plugins.gitstatus` and `config.plugins.gitstatus.color_icons` each 5 seconds for each file, this would add a bit overhead don't you think?

Reply:
The overhead is 0 (even considering that this function is called at every redraw, not only every 5 seconds, for each visible entry).
This is basically the standard way plugins override things. We would be in big trouble if this had a big overhead...

Reply:
Alright. It's still some additional checks to do but we should be fine for now. Let's go with your solution then.